netket/models/autoreg.py

```diff
@@ -90,8 +104,11 @@ def setup(self):
             for i in range(self.layers)
         ]
 
-    def conditionals(self, inputs: Array, cache: PyTree) -> Tuple[Array, PyTree]:
-        return _conditionals(self, inputs, cache)
+    def conditional(self, inputs: Array, index: int) -> Array:
```

Comment:
can I ask why you separated ```conditional``` and ```conditionals```?

Reply:
For fast autoregressive models, the logics of `conditional` and `conditionals` are very different. `conditionals` uses the whole inputs, and doesn't use the caches.

Also, `conditional` gives the expected output only if the correct caches are given in `variables`. Typically it should only be called successively with indices 0, 1, 2, ..., as in the autoregressive sampling procedure. Maybe we can make it a somewhat private interface, which should only be used by those who extend `ARNN`, not those who directly use prebuilt models.

Reply:
if the output of ```conditional``` is wrong (unless called in some special order) I would definitely not make that public, maybe since this is maybe an implementation detail of the fast version (if I understand correctly?) then it would be even better to just name it ```_fast_conditional``` or something private of this sort

Reply:
I agree, and it's also an implementation detail of AR samplers (if there are more AR samplers in future)